crates/wit-parser/src/ast/resolve.rs

```diff
@@ -470,22 +540,72 @@ impl<'a> Resolver<'a> {
         let mut iface_id_order = Vec::new();
         let mut world_id_order = Vec::new();
         for name in order {
-            match names.get(name).unwrap() {
-                ast::AstItem::Interface(_) => {
+            match names.get(name) {
+                Some(ast::AstItem::Interface(i)) => {
                     let id = self.alloc_interface(package_items[name]);
                     self.interfaces[id].name = Some(name.to_string());
+                    for item in &i.items {
```

Comment:
I think this loop and processing `nest` items in general would be best done during the elaboration of an interface to uniformly handle it next to other `interface` constructs